### PR TITLE
Use sitemap labels instead of name in the sitemap selection dialog

### LINKED ISSUE #595: Use sitemap labels instead of name in the sitemap selection dialog

### PR DESCRIPTION
**Pull request by mueller-ma**
Fix #595

The sitemap name is also shown in the settings. @FlorianSW Whats the best way to show the label there?

### CODE REVIEW

**Comment by FlorianSW** [comment]
You would need to save the label of the sitemap in another preference and make the "real" sitemap preference (with the name of the sitemap) hidden. Then, when the user clicks the "Clear sitemap" preference, both preferences need to be cleared.

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java [hunk 1]
```diff
@@ -821,13 +821,15 @@ public void onSuccess(Call call, int statusCode, Headers headers, byte[] respons
     private void showSitemapSelectionDialog(final List<OpenHABSitemap> sitemapList) {
         Log.d(TAG, "Opening sitemap selection dialog");
         final List<String> sitemapNameList = new ArrayList<String>();
+        final List<String> sitemapLabelList = new ArrayList<String>();
         for (int i = 0; i < sitemapList.size(); i++) {
             sitemapNameList.add(sitemapList.get(i).getName());
```
Comment (FlorianSW) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: is it really needed to have two lists here? Wouldn't it be possible to get rid of sitemapNameList? When accessing the sitemapNameList in the dialog, you could access sitemapList, get the element at the position of "item" and execute getName on that.

If we would've API Level 24 as the minimum supported, we could even use streams and wouldn't need any additional list here (at least not as a local variable).

**Comment by FlorianSW** [comment]
LGTM, however, there're merge conflicts.

**Comment by mueller-ma** [comment]
Going to wait for #526 before I merge, so I can be sure it doesnt brake #526

**Pull request merged**